### ISSUE TITLE
Add a switch to not download from Github in offline mode when boostrapping

### ISSUE THREAD

**Issue by cderv** [rstudio/renv#344]
I had an issue that freeze R on my RStudio servers because we are offline. 

Currently, `activate.R` has a boostraping mechanism that will try to install from Github if last version available in CRAN is not the correct one. 
https://github.com/rstudio/renv/blob/57eb53983d91809c520609a8cf0ffa564be4e7dd/inst/resources/activate.R#L139-L145

This would need to be deactivated in offline environment where we have only a package Manager or a CRAN mirror, at least with a switch using an option or an environment variable configure on the servers. 

Also, how about bootstraping older version by installing from archive ? 

Thanks. 

----

The error in the servers 

```r
Failed to find installation of renv -- attempting to bootstrap...
* Downloading renv 0.8.2 ... Error in utils::download.file(url, destfile = destfile, mode = "wb", quiet = TRUE) : 
  impossible d'ouvrir l'URL 'https://api.github.com/repos/rstudio/renv/tarball/0.8.2'
De plus : Warning message:
In utils::download.file(url, destfile = destfile, mode = "wb", quiet = TRUE) :
  URL 'https://api.github.com/repos/rstudio/renv/tarball/0.8.2': status was 'Couldn't connect to server'
Warning message:
Failed to find an renv installation: the project will not be loaded.
Use `renv::activate()` to re-initialize the project. 
```

**Comment by kevinushey** [comment]
This has now been implemented -- `renv` will attempt to install from the CRAN archives before attempting to download from GitHub.

**Comment by ticarki** [comment]
Hello,

I face a similar issue when using renv in an offline session. The `renv/activate.R` script tries to access 'https://cloud.r-project.org' which results in a freeze of the RStudio session during activation/bootstrapping. I now manually update the hard-coded repository links in the `renv/activate.R:88` script to our RSPM CRAN mirror checkpoints. Then it works perfectly. @kevinushey is there a cleaner way to circumvent this? I saw that you also look for the `'renv.repos.cran'` option, should one set this value in the `.Rprofile` file?

Thank you!

**Comment by kevinushey** [comment]
You can set:

```
RENV_CONFIG_REPOS_OVERRIDE = <repoUrl>
```

See https://github.com/rstudio/renv/blob/master/R/bootstrap.R#L20-L49 for the associated logic.

**Comment by ticarki** [comment]
Hi @kevinushey,

I tried this solution and the bootstrapping works perfectly. However, we maintain multiple repositories and after restoring the project it is not possible to install packages from the other repositories, which are set in `options('repos')`. Is there any working workaround available? (#716 might be related)
